### ISSUE TITLE
icub-main devel configuration fail on Debian Stable

### ISSUE THREAD

**Issue by traversaro** [robotology/icub-main#666]
See https://github.com/robotology/robotology-superbuild/runs/805730684 .  

Error log; 
~~~
2020-06-25T02:12:04.4164767Z CMake Error at src/libraries/iCubDev/CMakeLists.txt:29 (install):
2020-06-25T02:12:04.4164887Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4164996Z   "iCubDev".
2020-06-25T02:12:04.4165059Z 
2020-06-25T02:12:04.4165105Z 
2020-06-25T02:12:04.4165434Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4165565Z   export Export set "iCubDev" not found.
2020-06-25T02:12:04.4165671Z Call Stack (most recent call first):
2020-06-25T02:12:04.4165784Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4165906Z   src/libraries/iCubDev/CMakeLists.txt:37 (icub_install_basic_package_files)
2020-06-25T02:12:04.4166022Z 
2020-06-25T02:12:04.4166067Z 
2020-06-25T02:12:04.4166176Z CMake Error at src/libraries/ctrlLib/CMakeLists.txt:62 (install):
2020-06-25T02:12:04.4166294Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4166404Z   "ctrlLib".
2020-06-25T02:12:04.4166464Z 
2020-06-25T02:12:04.4166508Z 
2020-06-25T02:12:04.4166836Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4166964Z   export Export set "ctrlLib" not found.
2020-06-25T02:12:04.4167069Z Call Stack (most recent call first):
2020-06-25T02:12:04.4167181Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4167286Z   src/libraries/ctrlLib/CMakeLists.txt:70 (icub_install_basic_package_files)
2020-06-25T02:12:04.4167362Z 
2020-06-25T02:12:04.4167406Z 
2020-06-25T02:12:04.4167514Z CMake Error at src/libraries/skinDynLib/CMakeLists.txt:41 (install):
2020-06-25T02:12:04.4167677Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4167793Z   "skinDynLib".
2020-06-25T02:12:04.4167841Z 
2020-06-25T02:12:04.4167902Z 
2020-06-25T02:12:04.4168230Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4168354Z   export Export set "skinDynLib" not found.
2020-06-25T02:12:04.4168460Z Call Stack (most recent call first):
2020-06-25T02:12:04.4168557Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4168679Z   src/libraries/skinDynLib/CMakeLists.txt:49 (icub_install_basic_package_files)
2020-06-25T02:12:04.4168758Z 
2020-06-25T02:12:04.4168804Z 
2020-06-25T02:12:04.4168908Z CMake Error at src/libraries/iKin/CMakeLists.txt:60 (install):
2020-06-25T02:12:04.4169023Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4169135Z   "iKin".
2020-06-25T02:12:04.4169182Z 
2020-06-25T02:12:04.4169241Z 
2020-06-25T02:12:04.4169566Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4169693Z   export Export set "iKin" not found.
2020-06-25T02:12:04.4169785Z Call Stack (most recent call first):
2020-06-25T02:12:04.4169899Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4170020Z   src/libraries/iKin/CMakeLists.txt:68 (icub_install_basic_package_files)
2020-06-25T02:12:04.4170093Z 
2020-06-25T02:12:04.4170139Z 
2020-06-25T02:12:04.4170245Z CMake Error at src/libraries/iDyn/CMakeLists.txt:32 (install):
2020-06-25T02:12:04.4170359Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4170502Z   "iDyn".
2020-06-25T02:12:04.4170551Z 
2020-06-25T02:12:04.4170610Z 
2020-06-25T02:12:04.4170940Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4171057Z   export Export set "iDyn" not found.
2020-06-25T02:12:04.4171161Z Call Stack (most recent call first):
2020-06-25T02:12:04.4171273Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4171394Z   src/libraries/iDyn/CMakeLists.txt:40 (icub_install_basic_package_files)
2020-06-25T02:12:04.4171467Z 
2020-06-25T02:12:04.4171512Z 
2020-06-25T02:12:04.4171845Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4171979Z CMake Error at src/libraries/learningMachine/CMakeLists.txt:75 (install):
2020-06-25T02:12:04.4172100Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4172210Z   "learningMachine".
2020-06-25T02:12:04.4172261Z 
2020-06-25T02:12:04.4172382Z 
2020-06-25T02:12:04.4172714Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4173770Z   export Export set "learningMachine" not found.
2020-06-25T02:12:04.4173904Z Call Stack (most recent call first):
2020-06-25T02:12:04.4174040Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4174194Z   src/libraries/learningMachine/CMakeLists.txt:83 (icub_install_basic_package_files)
2020-06-25T02:12:04.4174274Z 
2020-06-25T02:12:04.4174344Z 
2020-06-25T02:12:04.4174471Z CMake Error at src/libraries/perceptiveModels/CMakeLists.txt:34 (install):
2020-06-25T02:12:04.4174617Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4175271Z   "perceptiveModels".
2020-06-25T02:12:04.4175351Z 
2020-06-25T02:12:04.4175406Z 
2020-06-25T02:12:04.4175854Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4176107Z   export Export set "perceptiveModels" not found.
2020-06-25T02:12:04.4176243Z Call Stack (most recent call first):
2020-06-25T02:12:04.4176381Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4176532Z   src/libraries/perceptiveModels/CMakeLists.txt:42 (icub_install_basic_package_files)
2020-06-25T02:12:04.4176614Z 
2020-06-25T02:12:04.4176685Z 
2020-06-25T02:12:04.4176818Z CMake Error at src/libraries/actionPrimitives/CMakeLists.txt:22 (install):
2020-06-25T02:12:04.4176967Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4177084Z   "actionPrimitives".
2020-06-25T02:12:04.4177162Z 
2020-06-25T02:12:04.4177218Z 
2020-06-25T02:12:04.4177625Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4177783Z   export Export set "actionPrimitives" not found.
2020-06-25T02:12:04.4177920Z Call Stack (most recent call first):
2020-06-25T02:12:04.4178061Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4178210Z   src/libraries/actionPrimitives/CMakeLists.txt:30 (icub_install_basic_package_files)
2020-06-25T02:12:04.4178287Z 
2020-06-25T02:12:04.4178359Z 
2020-06-25T02:12:04.4178674Z --  [x]   Plugin: icubmod_canmotioncontrol (ENABLE_icubmod_canmotioncontrol)
2020-06-25T02:12:04.4179022Z --  [x]   Plugin: icubmod_cartesiancontrollerserver (ENABLE_icubmod_cartesiancontrollerserver)
2020-06-25T02:12:04.4179383Z --  [x]   Plugin: icubmod_cartesiancontrollerclient (ENABLE_icubmod_cartesiancontrollerclient)
2020-06-25T02:12:04.4179706Z --  [ ]   Plugin: icubmod_cfw2can (ENABLE_icubmod_cfw2can)
2020-06-25T02:12:04.4180022Z --  [x]   Plugin: icubmod_dragonfly2 (ENABLE_icubmod_dragonfly2)
2020-06-25T02:12:04.4180347Z --  [x]   Plugin: icubmod_dragonfly2raw (ENABLE_icubmod_dragonfly2raw)
2020-06-25T02:12:04.4180704Z -- /usr/lib/x86_64-linux-gnu/libraw1394.so/usr/lib/x86_64-linux-gnu/libdc1394.so
2020-06-25T02:12:04.4181020Z --  [ ]   Plugin: icubmod_ecan (ENABLE_icubmod_ecan)
2020-06-25T02:12:04.4181332Z --  [x]   Plugin: icubmod_sharedcan (ENABLE_icubmod_sharedcan)
2020-06-25T02:12:04.4181646Z --  [ ]   Plugin: icubmod_socketcan (ENABLE_icubmod_socketcan)
2020-06-25T02:12:04.4181943Z --  [ ]   Plugin: icubmod_esdsniffer (ENABLE_icubmod_esdsniffer)
2020-06-25T02:12:04.4182252Z --  [ ]   Plugin: icubmod_fakecan (ENABLE_icubmod_fakecan)
2020-06-25T02:12:04.4182565Z --  [ ]   Plugin: icubmod_bcbBattery (ENABLE_icubmod_bcbBattery)
2020-06-25T02:12:04.4182880Z --  [ ]   Plugin: icubmod_bmsBattery (ENABLE_icubmod_bmsBattery)
2020-06-25T02:12:04.4183219Z --  [x]   Plugin: icubmod_gazecontrollerclient (ENABLE_icubmod_gazecontrollerclient)
2020-06-25T02:12:04.4183564Z --  [x]   Plugin: icubmod_parametricCalibrator (ENABLE_icubmod_parametricCalibrator)
2020-06-25T02:12:04.4183921Z --  [x]   Plugin: icubmod_parametricCalibratorEth (ENABLE_icubmod_parametricCalibratorEth)
2020-06-25T02:12:04.4184313Z --  [ ]   Plugin: icubmod_pcan (ENABLE_icubmod_pcan)
2020-06-25T02:12:04.4184639Z --  [ ]   Plugin: icubmod_static_grabber (ENABLE_icubmod_static_grabber)
2020-06-25T02:12:04.4185049Z --  [x]   Plugin: icubmod_xsensmtx (ENABLE_icubmod_xsensmtx)
2020-06-25T02:12:04.4185344Z --  [x]   Plugin: icubmod_skinWrapper (ENABLE_icubmod_skinWrapper)
2020-06-25T02:12:04.4185657Z --  [x]   Plugin: icubmod_canBusAnalogSensor (ENABLE_icubmod_canBusAnalogSensor)
2020-06-25T02:12:04.4185975Z --  [ ]   Plugin: icubmod_canBusDoubleFTSensor (ENABLE_icubmod_canBusDoubleFTSensor)
2020-06-25T02:12:04.4186303Z --  [x]   Plugin: icubmod_canBusVirtualAnalogSensor (ENABLE_icubmod_canBusVirtualAnalogSensor)
2020-06-25T02:12:04.4186619Z --  [x]   Plugin: icubmod_canBusInertialMTB (ENABLE_icubmod_canBusInertialMTB)
2020-06-25T02:12:04.4186914Z --  [x]   Plugin: icubmod_canBusSkin (ENABLE_icubmod_canBusSkin)
2020-06-25T02:12:04.4187226Z --  [x]   Plugin: icubmod_embObjMotionControl (ENABLE_icubmod_embObjMotionControl)
2020-06-25T02:12:04.4187580Z --  [x]   Plugin: icubmod_embObjSkin (ENABLE_icubmod_embObjSkin)
2020-06-25T02:12:04.4187888Z --  [ ]   Plugin: icubmod_embObjAnalogSensor (ENABLE_icubmod_embObjAnalogSensor)
2020-06-25T02:12:04.4188191Z --  [x]   Plugin: icubmod_embObjStrain (ENABLE_icubmod_embObjStrain)
2020-06-25T02:12:04.4188497Z --  [x]   Plugin: icubmod_embObjFTsensor (ENABLE_icubmod_embObjFTsensor)
2020-06-25T02:12:04.4188793Z --  [x]   Plugin: icubmod_embObjMais (ENABLE_icubmod_embObjMais)
2020-06-25T02:12:04.4189091Z --  [ ]   Plugin: icubmod_embObjMultiEnc (ENABLE_icubmod_embObjMultiEnc)
2020-06-25T02:12:04.4189398Z --  [x]   Plugin: icubmod_embObjInertials (ENABLE_icubmod_embObjInertials)
2020-06-25T02:12:04.4189694Z --  [x]   Plugin: icubmod_embObjIMU (ENABLE_icubmod_embObjIMU)
2020-06-25T02:12:04.4190022Z --  [x]   Plugin: icubmod_embObjVirtualAnalogSensor (ENABLE_icubmod_embObjVirtualAnalogSensor)
2020-06-25T02:12:04.4190318Z --  [ ]   Plugin: icubmod_imuST_M1 (ENABLE_icubmod_imuST_M1)
2020-06-25T02:12:04.4231702Z --  [x]   Plugin: icubmod_imuFilter (ENABLE_icubmod_imuFilter)
2020-06-25T02:12:04.4232011Z --  [ ]   Plugin: icubmod_embObjPSC (ENABLE_icubmod_embObjPSC)
2020-06-25T02:12:04.4232136Z CMake Error at src/libraries/optimization/CMakeLists.txt:38 (install):
2020-06-25T02:12:04.4232264Z   install TARGETS given no ARCHIVE DESTINATION for static library target
2020-06-25T02:12:04.4232378Z   "optimization".
2020-06-25T02:12:04.4232431Z 
2020-06-25T02:12:04.4232494Z 
2020-06-25T02:12:04.4232830Z CMake Error at /__w/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:670 (export):
2020-06-25T02:12:04.4232963Z   export Export set "optimization" not found.
2020-06-25T02:12:04.4233068Z Call Stack (most recent call first):
2020-06-25T02:12:04.4233169Z   conf/iCubInstallBasicPackageFiles.cmake:38 (install_basic_package_files)
2020-06-25T02:12:04.4233292Z   src/libraries/optimization/CMakeLists.txt:46 (icub_install_basic_package_files)
2020-06-25T02:12:04.4233374Z 
2020-06-25T02:12:04.4233420Z 
2020-06-25T02:12:04.4233640Z --  +++ tool compiling canLoaderLib
2020-06-25T02:12:04.4233869Z --  +++ tool compiling canLoader-console
2020-06-25T02:12:04.4234093Z --  +++ tool compiling ethLoaderLib
2020-06-25T02:12:04.4234318Z --  +++ tool compiling strainCalibLib
2020-06-25T02:12:04.4234657Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4235010Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4235361Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4235690Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4236194Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4236539Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4236891Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4237242Z -- Found YARP: /__w/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.3.103+20200623.4+git1ae7811f3")
2020-06-25T02:12:04.4237475Z -- Installing cmake scripts
2020-06-25T02:12:04.4237709Z -- Configuring incomplete, errors occurred!
~~~

On Debian Buster the CMake  used is 3.13 (see https://packages.debian.org/buster/cmake) so this could be relate to that.

**Comment by traversaro** [comment]
The problem is probably in all the `install` invocation in https://github.com/robotology/icub-main/pull/652/files#diff-743c289c8bd401436bb8cf793fb84942R62 that lack specification for `RUNTIME DESTINATION` and `ARCHIVE DESTINATION` (see https://github.com/robotology/how-to-export-cpp-library/blob/master/src/LibTemplateCMake/CMakeLists.txt#L86). This is not a problem in CMake >= 3.14 as in that case a default for that argument has been specified (see https://cmake.org/cmake/help/v3.14/command/install.html#installing-targets), but it is compulsory to set  `RUNTIME DESTINATION` and `ARCHIVE DESTINATION` to remain compatible with CMake 3.13 .

**Comment by pattacini** [comment]
@Nicogene fixed this in #667 👍
